### PR TITLE
NextJS: Fix v13 `next/link`

### PR DESCRIPTION
**Pull request by kylegach**
Issue: https://github.com/vercel/next.js/issues/42621 #19813

## What I did

Applied the internal NEXT env var in the Storybook env

## How to test

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

**Comment by kylegach** [comment]
These CI failures seem unrelated?

**Review by valentinpalkovic — APPROVED** [review]

**Comment by kylegach** [comment]
At Yann's suggestion, I tested in `next@12`. It appears the env var _does_ have an effect in that version, so I updated this to only apply to `next@13+`.

File: code/frameworks/nextjs/src/config/webpack.ts [hunk 1]
```diff
@@ -24,63 +22,21 @@ export const configureConfig = async ({
...
+  const definePluginConfig: Record<string, any> = {
+    // this mimics what nextjs does client side
+    // https://github.com/vercel/next.js/blob/57702cb2a9a9dba4b552e0007c16449cf36cfb44/packages/next/client/index.tsx#L101
+    'process.env.__NEXT_RUNTIME_CONFIG': JSON.stringify({
+      serverRuntimeConfig: {},
+      publicRuntimeConfig: nextConfig.publicRuntimeConfig,
+    }),
+  };
+
+  if (semver.gte(version, '13.0.0') || nextConfig.experimental?.newNextLinkBehavior) {
```
[the remark this answers is not in the document]
Comment (yannbf) [reply code/frameworks/nextjs/src/config/webpack.ts:37]: > This logic won't work if someone is using next@13 and sets newNextLinkBehavior to false (which is the only expected behavior for next@13, as the default is undefined).
That is intended, unless I'm very wrong, I don't think one can opt out of it in Next.js 13.

> And, if you're up for it, someone could also use next@12 set newNextLinkBehavior to true.
That is also intended, as the flag is for Next.js@12.2 projects, though people could set it (I doubt they would) if they used 12.0 or 12.1, but this wouldn't do anything.

[the remark this answers is not in the document]
Comment (kylegach) [reply code/frameworks/nextjs/src/config/webpack.ts:37]: > I don't think one can opt out of it in Next.js 13

Yeah, I understand why you'd think that. They don't document it anywhere, but [this comment](https://github.com/vercel/next.js/pull/42623#discussion_r1025455619) on a PR to remove `next/link`'s reliance on that env var would indicate they expect it to work.

> That is also intended, as the flag is for Next.js@12.2 projects, though people could set it (I doubt they would) if they used 12.0 or 12.1, but this wouldn't do anything.

Fair enough. I don't feel strongly about it, and if someone raises an issue, we can always add support later.

**Comment by shilman** [comment]
@kylegach @valentinpalkovic there's a merge conflict here -- can one of you please resolve it?

**Pull request merged**